### PR TITLE
FEATURE: Composer AI helper

### PR DESCRIPTION
**Pull request by romanrizzi**
This change introduces a new composer button for the group members listed in the `ai_helper_allowed_groups` site setting.

Users can use chatGPT to review, improve, or translate their posts to English.

### CODE REVIEW

File: app/controllers/discourse_ai/ai_helper/assistant_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      before_action :ensure_can_request_suggestions
+
+      def suggest
+        raise Discourse::InvalidParameters.new(:text) if params[:text].blank?
+
+        if !DiscourseAi::AiHelper::OpenAiPrompt::VALID_TYPES.include?(params[:mode])
+          raise Discourse::InvalidParameters.new(:mode)
+        end
+
+        RateLimiter.new(current_user, "ai_assistant", 6, 3.minutes).performed!
```
Comment (xfalcox) [inline app/controllers/discourse_ai/ai_helper/assistant_controller.rb:17]: Maybe bypass for staff? Specially now that everyone will be playing with it

File: app/controllers/discourse_ai/ai_helper/assistant_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        hijack do
+          response = {
+            suggestions:
+              DiscourseAi::AiHelper::OpenAiPrompt.new.generate_and_send_prompt(
+                params[:mode],
+                params[:text],
+              ),
+          }
+
+          if params[:mode] === DiscourseAi::AiHelper::OpenAiPrompt::PROOFREAD
```
Comment (xfalcox) [inline app/controllers/discourse_ai/ai_helper/assistant_controller.rb:28]: We should move this away from the controller and into lib next.

**Comment by xfalcox** [comment]
Looks great! Let's merge and let people test.

Next, I want to make the list of prompts completely data driven, so we can add / remove prompts from the database instead of hard coding it across various files. That will allow us to iterate quickly on the obscene amount of prompts suggestions we will receive.

File: config/settings.yml [hunk 3]
```diff
@@ -87,3 +87,14 @@ plugins:
 
   ai_openai_api_key:
     default: ""
+
+  composer_ai_helper_enabled:
```
Comment (xfalcox) [inline config/settings.yml:91]: We need a `composer_ai_allowed_in_pm: false` and the logic for it before merging.

**Review by xfalcox — CHANGES_REQUESTED** [review]
Only disabling it from PMs is a blocker, others are follow ups

File: app/controllers/discourse_ai/ai_helper/assistant_controller.rb (hunk 1 above)
Comment (romanrizzi) [reply app/controllers/discourse_ai/ai_helper/assistant_controller.rb:17]: Staff is bypassed by default unless specified otherwise

**Review by xfalcox — APPROVED** [review]

**Pull request merged**